crates/grafbase-sdk/wit/since_0_17_0/shared-context.wit

```diff
@@ -0,0 +1,11 @@
+interface shared-context {
+    use event-queue.{event-queue};
+
+    /// A shared context resource that provides access to tracing and event handling capabilities
+    resource shared-context {
+        /// Returns the request trace identifier as a string
+        trace-id: func() -> string;
+        /// Returns the request event queue
+        event-queue: func() -> event-queue;
```

Comment:
I don't think I would have exposed the event queue as is. This makes it look like you can also pop from anywhere. Would have added a `register-event` function and for the `on_response` would have actually provided a list of events. Since you need to explicitly register the events you're interested in, no need to pop them one by one. Well maybe not the most efficient design to allocate a vec all the time though, so well don't know for the `on_response` side

Reply:
It is not public for the user.

Reply:
The WIT is as public as the SDK itself. We just don't offer support for it